Sources/UI/Device/Device.swift

```diff
@@ -0,0 +1,405 @@
+//
+//  Device.swift
+//  RoverUI
+//
+//  Created by Andrew Clunis on 2018-11-20.
+//  Copyright © 2018 Rover Labs Inc. All rights reserved.
+//
+
+import Foundation
+import os
+import UIKit
+import UserNotifications
+
+class Device {
+    
+    let userDefaults: UserDefaults
```

Comment:
Can we remove the injected reference to `UserDefaults`, `JSONEncoder` and `JSONDecoder`? I've found the explicit rule of "every dependency must be injected" only adds to complexity and provides zero value. E.g. there is no reason to ever use any instance of `UserDefaults` other than `UserDefaults.standard`. Singletons have their drawbacks but it is the "iOS way" as Apple uses it _everywhere_.

Reply:
If you were to apply this pattern consistently in this class, you'd need to inject a ton more instances to avoid using their singletons. E.g. `UIDevice.current`, `Locale.current`, `Bundle.main` etc. It makes the constructor and assembler unnecessarily complex and doesn't add any value or additional configurability as each computed property can be overridden if needed.